src/Spago/Command/Fetch.purs

```diff
@@ -286,7 +291,7 @@ writeNewLockfile reason allTransitiveDeps = do
     processPackage :: LockfileBuilderResult -> Tuple PackageName (Tuple PackageName Package) -> Spago (FetchEnv a) LockfileBuilderResult
     processPackage result (Tuple workspacePackageName (Tuple dependencyName dependencyPackage)) = do
       (packageDependencies :: Array PackageName) <- (Array.fromFoldable <<< Map.keys <<< fromMaybe Map.empty)
-        <$> getPackageDependencies dependencyName dependencyPackage
+        <$> getPackageDependencies dependencyName dependencyPackage Map.empty
```

Comment:
It seems feasible to populate the map here, but OTOH it doesn't seem like it would save much time since the lock file is generated rarely.